js/wp-seo-admin.js

```diff
@@ -244,3 +244,73 @@ function wpseo_add_fb_admin() {
 		}
 	);
 }
+
+function wpseo_recalculate_scores(current_page) {
+	var total_count    = parseInt( jQuery( '#wpseo_count_total').text() );
+	var count_element  = jQuery( '#wpseo_count' );
+	var progress_bar   = jQuery( '#wpseo_progressbar' );
+
+	// Reset the count element and the progressbar
+	count_element.text( 0 );
+	progress_bar.progressbar( { 'value' : 0 } );
+
+	var update_count = function( total_posts ) {
+		var current_value = count_element.text();
+		var new_value     = parseInt( current_value ) + total_posts;
+		var new_width     = new_value * (100 / total_count);
+
+		progress_bar.progressbar( 'value', new_width );
+
+		count_element.html( new_value );
+	};
+
+	var calculate_score = function( post ) {
```

Comment:
I would call this function `save_recalculated_score(s)`, that way it is more clear what it is doing. Now it seems like `calculate_score` calculates the score for 1 post and `calculate_scores` for multiple posts, but that is not what the code is doing.